packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItem.js

```diff
@@ -85,14 +87,28 @@ export const ConditionBuilderItem = ({
     }
   }, [contentRef, open]);
 
+  const eligibleToClose = () => {
+    if (
+      event?.target?.closest?.(`.${carbonPrefix}--date-picker__calendar`) ||
```

Comment:
missing the `event` in the function definition

Reply:
Carbon is not providing the `event` for `onRequestClose` callback. But i refined the approach based on  Matt suggested.